src/hotspot/share/runtime/arguments.cpp

```diff
@@ -1180,7 +1180,7 @@ bool Arguments::process_settings_file(const char* file_name, bool should_exist,
   char quote_c        = 0;
   bool result         = true;
 
-  int c = getc(stream);
+  char c = checked_cast<char>(getc(stream));
```

Comment:
I think this only works if "char" is signed, because EOF is -1.  What I meant was:
int c0 = getc(stream);
while (c0 != EOF ...) {
  char c = checked_cast<char>(c0);
  ...
  c0 = getc(stream);

Reply:
I don't see this as any cleaner than having the checked_cast in the place where we need to store a char.